Uploaded all files and fixed errors

DH3773 opened the pull request:
Uploaded all of the theory files and fixed all of the syntax errors.

DH3773 committed:
Uploaded all files and fixed errors

alyssa-blair commented on Theory/Note.cs:
```diff
@@ -0,0 +1,71 @@
+    {
+        var index = Array.IndexOf(_allNotes, _letter);
+        
+        if(index == 0) 
```
Everything looks good! Could update this to a modular calculation ((index + 11) % 12) if you wanted to but up to you!

alyssa-blair commented on Theory/Note.cs:
```diff
@@ -0,0 +1,71 @@
+    {
+        var index = Array.IndexOf(_allNotes, _letter);
+
+        if (index == 11)
```
could also change this one to modular ((index + 1) % 12)

wrighted replied:
Would prefer to see dynamic implementation:

```
int index = Array.IndexOf(_allNotes, _letter);
        
int newIndex = index >= _allNotes.Length - 1 ? 0 : index + 1;
return new Note(_allNotes[newIndex]);
```

wrighted commented on Theory/Chords.cs:
```diff
@@ -0,0 +1,366 @@
+namespace Fretty.Theory;
+
+public class Chords
```
This class can be `static` I think.

wrighted commented on Theory/GuitarString.cs:
```diff
@@ -0,0 +1,47 @@
+
+    // Returns the note at the given fret
+    // where fret 0 is the open string 
+    public Note at_fret(int fret) {
```
`AtFret`

wrighted commented on Theory/GuitarString.cs:
```diff
@@ -0,0 +1,47 @@
+    // Returns the array of Notes that 
+    // represents all of the notes on 
+    // the Guitar_String
+    public Note[] to_notes_array() {
```
`AsNotes`

wrighted commented on Theory/Scales.cs:
```diff
@@ -0,0 +1,187 @@
+namespace Fretty.Theory;
+
+// Class holding all scales and scale-related functions
+public class Scales
```
This class can be `static`.

wrighted commented on Theory/Scales.cs:
```diff
@@ -0,0 +1,187 @@
+
+
+    //TODO, change this to use Chord class? Or maybe just have a function in chord class that returns array of notes
+    public static List<ScaleName> ScalesContainingChords(Note[][] chords)
```
I like the comments here, nice work.

wrighted commented on Theory/Song.cs:
```diff
@@ -0,0 +1,30 @@
+namespace Fretty.Theory;
+
+public class Song(string songName, Key songKey, Chords.ChordName[] songChords)
```
If you add:

`using Chords;`

above this type can become:

`ChordName[]`

wrighted commented on Theory/Tuning.cs:
```diff
@@ -0,0 +1,25 @@
+        }
+        else
+        {
+            //Throw exception
```
I don't think this comment adds much.

wrighted commented on Theory/Voicing.cs:
```diff
@@ -0,0 +1,36 @@
+    // style, with {Chord: Voicing1, Voicing2, etc}
+    private int[] _positions;
+
+    public Voicing(int[] fretPositions, int[] positions)
```
What's the intended use of a Voicing? ie) What is parameter `int[] positions`s role?

wrighted commented on TheoryTests/NoteTest.cs:
```diff
@@ -0,0 +1,16 @@
+// namespace Fretty.TheoryTests;
```
Still working on adding testing framework.

wrighted reviewed (commented):
Just a few very minor changes I want to see, seems like a fairly robust object approach at representing guitar logic, nice work.

DH3773 committed:
Added functionality to multiple classes

Added a bunch of methods and finished some TODO stuff. Also added more cross-compatibility/cross-talk between classes

wrighted reviewed (approved):
Let's merge this so we can start testing end-to-end.

DH3773 committed:
Changed some code based on comments
